components/page/DevPageHeader.tsx

```diff
@@ -0,0 +1,71 @@
+import { Devfile } from 'custom-types'
+
+import {
+    Badge,
+    Card,
+    CardBody
+  } from '@patternfly/react-core'
+
+/**
+ * props for devpage metadata header 
+ * @param devfile - devfile index information
+ * @param metadata - metadata from 
+ */
+interface Props {
+    devfile:Devfile,
+    metadata?:Record<string, string>
+}
+// fake image must remove and replace with real image import 
+const FakeImage = () =>{
+    return (
+        <div style={{backgroundColor:"#000000",height:"70px", width:"70px"}}></div>
+    )
+}
+/**
+ * header that displays basic information and metadata information for respective devfile
+ * @param devfile - index information for devfie
+ * @param metadata - metadata information from devfile yaml
+ */
+const DevPageHeader = ({devfile, metadata}:Props) => {
+    const metaInclude = new Set(['projectType', 'version', 'type', 'language' ]) // types to include in metadata from index
```

Comment:
From my understanding of a Set vs Array you should be using an array here because it is faster since array elements are next to each other while set elements are not. https://medium.com/front-end-weekly/es6-set-vs-array-what-and-when-efc055655e1a

Reply:
The reason I say this is because I think people are more knowledgeable about array methods so its easier to read.

Reply:
i used sets to indicate that ordering doesn't matter and since it prevents duplication... also arrays are slower depending on # of elements since most methods iterate through each element until the desired element is found while sets use hashing 
but its likely true that for a small number of elements (in this case very smol lol) computing the hash takes longer then checking if it exists among an array of four which i didn't think about so thanks for the pointer:)
anyways, i changed it to an array so resolved